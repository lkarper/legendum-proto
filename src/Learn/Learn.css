@import url('https://fonts.googleapis.com/css2?family=Cinzel+Decorative:wght@400;700&family=Crimson+Text&display=swap');

.Learn__container {
    position: relative;
    display: flex;
    flex-direction: column;
    align-items: center;
    min-height: 100%;
    width: 100%;
}

.Learn__loading {
    font-size: 20px;
}

@media screen and (min-width: 1000px) {
    .Learn__container {
        width: 800px;
    }

    .Learn__loading {
        font-size: 40px;
    }
}