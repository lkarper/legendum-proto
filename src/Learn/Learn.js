import React, { useState, useEffect } from 'react';
import ExercisesService from '../services/exercises-service';
import LearnPage from '../LearnPage/LearnPage';
import LoadingSpinner from '../LoadingSpinner/LoadingSpinner';
import './Learn.css';

const Learn = (props) => {
    const { chapt } = props.match.params;

    const [pages, setPages] = useState([]);
    const [page, setPage] = useState(1);
    const [showLoading, setShowLoading] = useState(false);
    const [apiError, setApiError] = useState(false);

    useEffect(() => {
        window.scrollTo(0, 0);
        setShowLoading(true);
        ExercisesService.getExercisesLearnByChapter(chapt)
            .then(data => {
                setApiError(false);
                data.sort((a, b) => a.page - b.page);
                setPages(data);
                setShowLoading(false);
            })
            .catch(error => {
                setApiError(true);
                setShowLoading(false);
                window.scrollTo(0, document.querySelector('.Learn__alert-div').offsetTop - document.querySelector('.Header__header').offsetHeight);
                console.log('error', error);
            });
    }, [chapt]);

    return (
        <>
            {pages.length !== 0 && 
                <div className='Learn__container'>
                    <LearnPage
                        data={{
                            pages,
                            page,
                            chapt,
                            setPage,
                        }}
                    />
                </div>
            }
            {showLoading && <LoadingSpinner />}
            <div
                className='Learn__alert-div'
                role='alert'
            >
                {apiError && <p>Error: Looks like something went wrong. Check the url and your connection and try again.</p>}
            </div>
        </>
    );
}

export default Learn;
