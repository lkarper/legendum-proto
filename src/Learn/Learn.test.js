import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Learn from './Learn';

describe('Learn component', () => {
    it('renders the UI without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Learn />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const wrapper = mount(<Learn />);
        expect(toJSON(wrapper)).toMatchSnapshot();
        wrapper.unmount();
    });
});
