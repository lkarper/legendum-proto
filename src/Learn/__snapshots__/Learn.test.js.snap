// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Learn component renders the UI as expected 1`] = `
<Learn
  match={
    Object {
      "params": Object {
        "chapt": "",
      },
    }
  }
>
  <LoadingSpinner>
    <div
      className="LoadingSpinner__loading-container"
    >
      <FontAwesomeIcon
        aria-label="Loading."
        className="LoadingSpinner__spinner"
        icon={
          Array [
            "fa",
            "spinner",
          ]
        }
        spin={true}
      >
        <i
          className="fa"
        />
      </FontAwesomeIcon>
    </div>
  </LoadingSpinner>
  <div
    className="Learn__alert-div"
    role="alert"
  />
</Learn>
`;
